compute/metadata/metadata.go

```diff
@@ -304,8 +306,20 @@ func (c *Client) getETag(suffix string) (value, etag string, err error) {
 	}
 	req.Header.Set("Metadata-Flavor", "Google")
 	req.Header.Set("User-Agent", userAgent)
-	res, err := c.hc.Do(req)
-	if err != nil {
+	var res *http.Response
+	retryer := newRetryer()
```

Comment:
This appears to be race free because each call creates its own retryer. Are there any other concerns?

Reply:
correct, with how our gax backoff works today I think you always need to return a new one per call so that the backoff can work accordingly.